lib/presentation/screens/profile_screen.dart

```diff
@@ -120,18 +128,79 @@ class _ProfileScreenState extends State<ProfileScreen> {
     );
   }
 
+  // Updated method to navigate to Event Details Screen
   void _showEventDetails(int index) {
-    showDialog(
-      context: context,
-      builder: (BuildContext context) {
-        return AlertDialog(
-          title: const Text(ProfileConstants.eventDetailsTitle),
-          content: Text(_pastEvents[index]),
-          actions: [
-            TextButton(onPressed: () => Navigator.of(context).pop(), child: const Text(ProfileConstants.closeButton)),
-          ],
-        );
-      },
+    // Create an Event object based on the past event
+    final eventTitle = _pastEvents[index].split(' - ')[0];
+    final eventDate = _pastEvents[index].split(' - ')[1];
+
+    // Parse the date (you might want to store actual DateTime objects)
+    DateTime parsedDate;
+    try {
+      parsedDate = DateTime.parse(eventDate.replaceAll(',', '').replaceAll(' ', '-'));
```

Comment:
this same logic can be reused later, let's create a function nd move it to that common folder